src/pages/participants.js

```diff
@@ -4,188 +4,611 @@ import { FaTwitter } from "react-icons/fa";
 import Helmet from "../components/helmet";
 import Navbar from "../components/navbar";
 
-export default () => (
-  <div>
-    <Helmet />
-    <Navbar />
-    <section className="section">
-      <h1 className="title has-text-centered">Participant registration</h1>
-    </section>
-    <section className="hero is-primary is-bold is-medium">
-      <div className="hero-body">
-        <div className="container">
-          <h1 className="title has-text-centered">
-            We are sorry but the registration is not open yet!
-          </h1>
-          <div className="buttons is-centered">
-            <div className="button is-medium is-link is-in-hero">
-              <span className="icon">
-                <FaTwitter size="fa" />
-              </span>
-              <a
-                href="https://twitter.com/pylondiniumgir1/"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                Keep connected
-              </a>
-            </div>
-          </div>
-        </div>
-      </div>
-    </section>
-    <section className="hero is-primary is-bold is-medium is-hidden-touch is-hidden-tablet">
-      <div className="hero-body">
-        <div className="container">
-          <h1 className="title has-text-centered">The registration is open!</h1>
-        </div>
-      </div>
-    </section>
-    <section className="section is-hidden-touch is-hidden-tablet">
-      <div className="columns is-centered">
-        <div className="column is-half">
+function Request(form) {
```

Comment:
Given how this is used, it seems to me it's using an old school JS patterns in which the object-like nature of JS functions is abused. In my view, this would have a much clearer interface if it were an actual class. This way, `.validate()` would just be a method, `.errors` would be a property (https://javascript.info/class#class-properties, but beware of browser support), and the `inputErrors` object would be initialised by the constructor.